### PR TITLE
Introduce error notification for failure of password reset 

### PR DESCRIPTION
**Pull request by Steffengreiner**
**Purpose of this PR**
This PR introduces an error notification shown to the user if the password reset fails

**How to Test**
In the Password Reset View, input an email address not stored in the database and click the submit button. 
The Password Reset

**Possible Improvements**
The Handler could try to distinct between a password reset failure caused by a domain error a passwort reset failure caused by an invalid email input reset. 
To do so we would need to introduce a dedicated PasswortResetException similiar to the UserRegistrationException which can then be parsed from the ApplicationResponse via the output interface to the Handler

### CODE REVIEW

File: webapp/src/main/java/life/qbic/views/login/passwordreset/PasswordResetHandler.java [hunk 1]
```diff
@@ -51,7 +66,6 @@ public void onPasswordResetSucceeded() {
 
     @Override
     public void onPasswordResetFailed() {
-        //todo
-        throw new NotImplementedException();
+        showPasswordResetFailedError();
```
Comment (sven1103) [inline webapp/src/main/java/life/qbic/views/login/passwordreset/PasswordResetHandler.java:69]: If we handle use case exceptions, i think we should do it with max satisfaction as possible to the user.

"The Password Reset has failed", "Please try again."

The user does not know, what they have done wrong. So i think we should increase transparency a little bit, and introduce methods in the output interface.

For example we can invoke a method `onUnknownEmailAddress()` if the email does not belong to any registered and active user account. We can further introduce a `onUnexpectedException()`, that will redirect the user to support@qbic.zendesk.com to handle more severe scenarios.

**Review by sven1103 — CHANGES_REQUESTED** [review]
Dear @Steffengreiner thank you for your contribution. I think if we introduce error messages, we should make the experience for the user as good as possible (given that sth failed, will already lead to increased frustration levels).

I have made some suggestions to improve the handling.

File: webapp/src/main/java/life/qbic/views/login/passwordreset/PasswordResetHandler.java [hunk 2]
```diff
@@ -3,55 +3,93 @@
...
+    for (RuntimeException failure : response.failures()) {
+      if (failure instanceof EmailAddress.EmailValidationException) {
+        showPasswordResetFailedError(failure.getMessage(), "Please provide a valid email address.");
+      }
+      else if (failure instanceof UserNotFoundException) {
+        showPasswordResetFailedError(failure.getMessage(), "No user registered for the provided email.");
+      }
+      else if (failure instanceof UserRegistrationService.UserNotActivatedException) {
+        showPasswordResetFailedError(
+            failure.getMessage(), "Activate your account to reset the password.");
```
Comment (sven1103) [inline webapp/src/main/java/life/qbic/views/login/passwordreset/PasswordResetHandler.java:88]:
```suggestion
            failure.getMessage(), "Please activate your account first to reset the password.");
```

File: webapp/src/main/java/life/qbic/views/login/passwordreset/PasswordResetHandler.java [hunk 3]
```diff
@@ -3,55 +3,93 @@
...
-        //todo
-        throw new NotImplementedException();
+  @Override
+  public void onPasswordResetFailed(ApplicationResponse response) {
+    for (RuntimeException failure : response.failures()) {
+      if (failure instanceof EmailAddress.EmailValidationException) {
+        showPasswordResetFailedError(failure.getMessage(), "Please provide a valid email address.");
+      }
+      else if (failure instanceof UserNotFoundException) {
+        showPasswordResetFailedError(failure.getMessage(), "No user registered for the provided email.");
```
Comment (sven1103) [inline webapp/src/main/java/life/qbic/views/login/passwordreset/PasswordResetHandler.java:84]:
```suggestion
        showPasswordResetFailedError(failure.getMessage(), "No user with the provided email address is known.");
```

**Review by sven1103 — CHANGES_REQUESTED** [review]
Only some minor phrasing suggestions

File: webapp/src/main/java/life/qbic/views/login/passwordreset/PasswordResetHandler.java [hunk 4]
```diff
@@ -3,55 +3,93 @@
...
 
-    @Override
-    public void onPasswordResetFailed() {
-        //todo
-        throw new NotImplementedException();
+  @Override
+  public void onPasswordResetFailed(ApplicationResponse response) {
+    for (RuntimeException failure : response.failures()) {
+      if (failure instanceof EmailAddress.EmailValidationException) {
+        showPasswordResetFailedError(failure.getMessage(), "Please provide a valid email address.");
```
Comment (sven1103) [inline webapp/src/main/java/life/qbic/views/login/passwordreset/PasswordResetHandler.java:81]: I think we should not use the exception message here, and remain in control what we display to the user. Based on the exception type, we can define that easily.

File: webapp/src/main/java/life/qbic/views/login/passwordreset/PasswordResetHandler.java [hunk 5]
```diff
@@ -3,55 +3,93 @@
...
+      }
+      else if (failure instanceof UserNotFoundException) {
+        showPasswordResetFailedError(failure.getMessage(), "No user with the provided email address is known.");
```
Comment (sven1103) [inline webapp/src/main/java/life/qbic/views/login/passwordreset/PasswordResetHandler.java:84]: same here, i think defining the message title explicit is more robust.

File: webapp/src/main/java/life/qbic/views/login/passwordreset/PasswordResetHandler.java [hunk 6]
```diff
@@ -3,55 +3,93 @@
...
+      }
+      else if (failure instanceof UserRegistrationService.UserNotActivatedException) {
+        showPasswordResetFailedError(
+            failure.getMessage(), "Please activate your account first to reset the password.");
```
Comment (sven1103) [inline webapp/src/main/java/life/qbic/views/login/passwordreset/PasswordResetHandler.java:88]: same here

File: webapp/src/main/java/life/qbic/views/login/passwordreset/PasswordResetHandler.java (hunk 4 above)
Comment (jenniferboedker) [reply webapp/src/main/java/life/qbic/views/login/passwordreset/PasswordResetHandler.java:81]: I am ok with that. Will do

**Review by sven1103 — APPROVED** [review]
great !

**Pull request merged**